Gems/PhysX/Code/Source/Utils.cpp

```diff
@@ -70,8 +71,8 @@ namespace PhysX
 
         AZStd::pair<uint8_t, uint8_t> GetPhysXMaterialIndicesFromHeightfieldSamples(
             const AZStd::vector<Physics::HeightMaterialPoint>& samples,
-            const int32_t row, const int32_t col,
-            const int32_t numRows, const int32_t numCols)
+            const size_t col, const size_t row, 
+            const size_t numCols, const size_t numRows)
```

Comment:
question: I'm curious why the change from (row, col) to (col, row)? Not very important, but normally I'd expect (row, col), for example like when accessing elements on a matrix.

I guess the important thing is that it's consistent across all functions, otherwise it'll be very easy to make mistakes and pass the wrong order.

Reply:
While working on this PR, there were three things that caught my eye that caused me to do a pass on these APIs:
1. The modifySamples() call that I was adding from PhysX looks like this: `modifySamples (PxI32 startCol, PxI32 startRow, const PxHeightFieldDesc &subfieldDesc, bool shrinkBounds=false)` . I initially had a bug in my implementation because I was passing in (row, col) similar to our other existing APIs.
2. There were a few places that we're passing in variables named x/y or width/height in for row and column values. It looked really weird to pass them as (y, x) or (height, width).
3. I think there were a couple places that were using (col, row) already (or maybe I implemented them myself during this PR, causing the problem myself, I don't recall now), where I think the parameters of one were getting passed into a different API where the parameters needed to get a flipped order.

That said, even PhysX isn't consistent within their API, because there's also this: `PxHeightFieldGeometry (PxHeightField *hf, PxMeshGeometryFlags flags, PxReal heightScale_, PxReal rowScale_, PxReal columnScale_)`

The other consistency pass I did at the same time was that we had a mix of usage between int32_t, uint32_t, and size_t. This was leading to some casts back and forth for no particular reason. I semi-arbitrarily picked size_t as the one to standardize on, since this matched many of the other Terrain APIs.

Reply:
Thanks for the detailed answer. It makes sense. This PR is making it a lot more consistent, which is a great change. Cheers.